lib/FusionInventory/Agent/Task/Inventory/Generic/Batteries/Upower.pm

```diff
@@ -0,0 +1,102 @@
+package FusionInventory::Agent::Task::Inventory::Generic::Batteries::Upower;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Generic;
+
+my $command = 'upower';
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{battery};
+    return canRun($command);
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+
+    my @batteries = _getBatteriesFromUpower(%params);
+
+    return unless @batteries;
+
+    for my $batt (@batteries) {
+        $inventory->addEntry(
```

Comment:
Don't you should also use merging process here as nothing seems to guaranty Upower module passed before dmidecode one ? Then a merging callback telling who is mandatory could be useful in that case.